core/frontend/src/main/scala/io/udash/utils/HttpResponse.scala

```diff
@@ -0,0 +1,26 @@
+package io.udash.utils
+
+import com.avsystem.commons.misc.AbstractCase
+import org.scalajs.dom.Document
+import org.scalajs.dom.raw.XMLHttpRequest
+
+sealed trait HttpResponse {
```

Comment:
I think you don't need this sealed trait, just create a single case class. You can also move it to `FileUploader` companion object. 

We should add information about `status` code and provide access to response headers (`getResponseHeader`) from XHR.